internal/otel/otel.go

```diff
@@ -0,0 +1,146 @@
+package otel
```

Comment:
Why we have put this under internal, would this component be useful for other projects as well?

Reply:
True. Although I'm a fan of starting with everything private and opening it up as needed this might be a good candidate for making it public from the start.

Reply:
This doesn't seem to have any dependency on rudder-server's other packages. Would it make sense to use a separate repository for it so that other projects don't have to depend on rudder-server?

Reply:
A separate repo just for this I'd say it is overkilling. Unless we're planning to do a collection of these "utils". At Netdata we used to have a versioned `cloud-kit` repository with all these little independent packages used by different µservices and it was quite useful.

Reply:
For now I moved the `otel` package in the root folder.